packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -207,21 +220,38 @@ function* doStartPurchaseContentFlow({
         .div(BN_USDC_CENT_WEI)
         .toNumber()
       yield* put(buyUSDC())
-      yield* put(
-        onrampOpened({
-          provider: USDCOnRampProvider.STRIPE,
-          purchaseInfo: {
-            desiredAmount: balanceNeededCents
-          }
+      let result: { succeeded?: any; failed?: any; canceled?: any } | null =
+        null
+      if (isBuyUSDCViaSolEnabled) {
+        yield* put(
+          buyCryptoViaSol({
+            // expects "friendly" amount, so dollars
+            amount: balanceNeededCents / 100.0,
```

Comment:
this is such a big gotcha, saw this on the usdc side too

Reply:
Agreed, this is very error prone. I think we should:
* Make the naming convention very clear (so here it should be `amountDollars`)
* Maybe choose one representation to use in the majority of places and have code which needs it in a different format convert it (so in this case, `buyCryptoViaSol` would still accept it in cents and then convert it internally to dollars since it's just because Stipe wants it in dollars)